src/aiodynamo/client.py

```diff
@@ -315,61 +399,68 @@ def table(self, name: TableName) -> Table:
         key: Dict[str, Any],
         *,
         return_values: ReturnValues = ReturnValues.none,
-        condition: ConditionBase = None
+        condition: Condition = None,
     ) -> Union[None, Item]:
         key = py2dy(key)
         if not key:
             raise EmptyItem()
 
+        payload = {
+            "TableName": table,
+            "Key": key,
+            "ReturnValues": return_values.value,
+        }
+
         if condition:
-            (
-                condition_expression,
-                expression_attribute_names,
-                expression_attribute_values,
-            ) = ConditionExpressionBuilder().build_expression(condition)
-        else:
-            condition_expression = (
-                expression_attribute_names
-            ) = expression_attribute_values = None
-        resp = await self.core.delete_item(
-            **clean(
-                TableName=table,
-                Key=key,
-                ReturnValues=return_values.value,
-                ConditionExpression=condition_expression,
-                ExpressionAttributeNames=expression_attribute_names,
-                ExpressionAttributeValues=expression_attribute_values,
-            )
-        )
+            params = Parameters()
+            payload["ConditionExpression"] = condition.encode(params)
+            payload["ExpressionAttributeNames"] = params.get_expression_names()
+            payload["ExpressionAttributeValues"] = params.get_expression_values()
+
+        resp = await self.send_request(action="DeleteItem", payload=payload)
         if "Attributes" in resp:
             return dy2py(resp["Attributes"], self.numeric_type)
 
         else:
             return None
 
-    async def delete_table(self, table: TableName):
-        await self.core.delete_table(TableName=table)
+    async def delete_table(
+        self, table: TableName, *, wait_for_disabled: Union[bool, WaitConfig] = False
+    ):
+        await self.send_request(action="DeleteTable", payload={"TableName": table})
+        if not isinstance(wait_for_disabled, WaitConfig):
+            wait_for_disabled = WaitConfig.default()
+        attempts = 0
+        while attempts < wait_for_disabled.max_attempts:
+            try:
+                await self.describe_table(table)
+            except TableNotFound:
+                return
+            attempts += 1
+            await asyncio.sleep(wait_for_disabled.retry_delay)
+        raise TableDidNotBecomeDisabled()
```

Comment:
some sort of generic async generator might work

pardon my pseudopython
```
some_generator(the_coro, max_attempts, delay):
attempts = 0
while attempts < max_attempts:
    yield the_coro()
    attempts += 1
    await asyncio.sleep(delay)
raise OutOfAttempts
```

in this case
```python3
try:
    async for state in some_generator(self.describe_table(table), wait_for_disabled.max_attempts, wait_for_disabled.retry_delay):
        pass
except TableNotFound:
    return
except OutOfAttempts:
    raise TableDidNotBecomeDisabled()
#Any other exception will be raised I assume?    
```

in other case the result can be compared with state and just stops the generator when desired state has been reached?

Reply:
did something generator based, but a little different. what do you think of d60102c ?